src/iop/denoiseprofile.c

```diff
@@ -1075,7 +1084,13 @@ static void process_nlmeans(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t
   // adjust to zoom size:
   const float scale = fmin(roi_in->scale, 2.0f) / fmax(piece->iscale, 1.0f);
   const int P = ceilf(d->radius * scale); // pixel filter size
-  const int K = ceilf(d->nbhood * scale); // nbhood
+  const int K = ceilf(DT_IOP_DENOISE_PROFILE_NBHOOD_NORMAL_SIZE * scale);
+
+  int coordinate_offsets[DT_IOP_DENOISE_PROFILE_NBHOOD_NORMAL_SIZE];
+  for(int i = 0; i < DT_IOP_DENOISE_PROFILE_NBHOOD_NORMAL_SIZE; i++)
+  {
+    coordinate_offsets[i] = i * i * d->nbhood + i;
```

Comment:
Ok, let me just try an assertion to check I'm reading this correctly.

- before nbhood was 7 and we are looking at the 7 neighborhood.
- now nbhood is 0, and the coordinate_offsets is "coordinate_offsets[v] = v" so we are also looking at the same 7 neighborhood.
- if nbhood is changed then we have 7 more spread points to look at (for catching coarse grain).

Are all this right?

If so, indeed we just break history with recent change and this is not an issue since it was only on a dev release. So waiting for your comment before merging.

Reply:
I'm stupid, I just saw your second comment explaining exactly that. So I have my answer, and it is good to see that I got it right :)

Reply:
Yes, you understand it right!
The problem with grids comes from the fact that we have grid gaps between the patches we consider: lines or columns without any patches, and lines or columns with a lot of patches. I updated the formula so that the patches are chosen in a way that makes the patches better distributed.